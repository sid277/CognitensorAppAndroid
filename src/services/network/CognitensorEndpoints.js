import axios from 'axios';
import CognitensorAsyncStorageService from '../asyncstorage/CognitensorAsyncStorageService';
import { CLIENT_USER_URL } from '../../constants';

class CognitensorEndpoints {
  api = async ({ url, method, dispatchReducer }) => {
    dispatchReducer({ type: 'API_FETCH_INIT' });
    const token = await CognitensorAsyncStorageService.getUserToken();
    const reqConfig = {
      method,
      url,
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    };

    // TODO: Check if device is online
    axios(reqConfig)
      .then((result) => {
        if (result.status === 200) {
          dispatchReducer({
            type: 'API_FETCH_SUCCESS',
            payload: result.data,
          });
        }
      })
      .catch((e) => {
        dispatchReducer({ type: 'API_FETCH_FAILURE' });
        console.warn(e);
      });
  };

  login = async (email, password, dispatchReducer) => {
    axios({
      method: 'post',
      url: `${CLIENT_USER_URL}/auth/login`,
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      data: JSON.stringify({
        email: email.toLowerCase().trim(),
        password: password,
      }),
    })
      .then(async (response) => {
        if (response.data.token) {
          await CognitensorAsyncStorageService.setUserToken(
            response.data.token,
          );
          dispatchReducer({
            type: 'SIGN_IN',
            token: response.data.token,
            user: response.data.user,
          });
        }
        if (response.statusText === 'error') {
          console.log(response);
        }
      })
      .catch((err) => {
        console.log(err);
      });
  };

  signOut = async (dispatchReducer) => {
    await CognitensorAsyncStorageService.removeUserToken();
    dispatchReducer({ type: 'SIGN_OUT' });
  };

  getUserDetails = async ({ dispatchReducer }) => {
    await this.api({
      url: `${CLIENT_USER_URL}/auth/user/data`,
      method: 'get',
      dispatchReducer,
    });
  };

  getDashboardList = async ({ dispatchReducer }) => {
    await this.api({
      url: `${CLIENT_USER_URL}/cogniviz/dashboard/titles`,
      method: 'get',
      dispatchReducer,
    });
  };
}

export default new CognitensorEndpoints();
