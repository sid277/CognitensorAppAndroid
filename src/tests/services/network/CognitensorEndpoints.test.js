import React from 'react';
import { mount } from 'enzyme';
import CognitensorEndpoints from '../../../services/network/CognitensorEndpoints';
import renderer from 'react-test-renderer';

test('should test CognitensorEndpoints component', () => {
const wrapper = CognitensorEndpoints;
expect(wrapper).toMatchSnapshot();
});