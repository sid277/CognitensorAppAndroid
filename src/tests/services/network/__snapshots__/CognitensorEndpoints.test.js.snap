// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test CognitensorEndpoints component 1`] = `
CognitensorEndpoints {
  "api": [Function],
  "getDashboardList": [Function],
  "getUserDetails": [Function],
  "login": [Function],
  "signOut": [Function],
}
`;
